### ISSUE TITLE
Problem with HDF5 since last commit on main 

### ISSUE THREAD

**Issue by paulwilhelmvlasov** [selalib/selalib#31]
Hey, since the last commit I get Linker-Errors connected to HDF5:

```
[ 35%] Linking Fortran executable ../../../../bin/test_distribution_function
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.a(H5Z.o): in function `H5Z__init_package.part.0':
(.text+0x6b5): undefined reference to `SZ_encoder_enabled'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.a(H5Zszip.o): in function `H5Z__filter_szip':
(.text+0xbb): undefined reference to `SZ_BufftoBuffDecompress'
/usr/bin/ld: (.text+0x147): undefined reference to `SZ_BufftoBuffCompress'
collect2: error: ld returned 1 exit status
make[2]: *** [src/data_structures/distribution_function/testing/CMakeFiles/test_distribution_function.dir/build.make:137: bin/test_distribution_function] Error 1
make[1]: *** [CMakeFiles/Makefile2:5284: src/data_structures/distribution_function/testing/CMakeFiles/test_distribution_function.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

When switching back to an older commit, the build process finishes successfully. Is this a problem with my build system or was there some change in Linking to HDF5 maybe?

Best regards,
Paul

**Comment by pnavaro** [comment]
Could you make a test Paul ? Sorry I removed too much lines last time.

**Comment by paulwilhelmvlasov** [comment]
Yes, now it compiles and I also can find the simulation-executables in the bin-folder, so all fine now apparently. Thanks! :)

**Comment by kkormann** [comment]
Thanks, @pnavaro, for helping out with these HDF5 problems. Unfortunately, I got another one when I try to install selalib on a local cluster.  I am not sure if this is a problem with the HDF5 in our cmake or of the HDF5 configuration on the cluster. Here is the error message:

CMake Error at .../cmake/3.24.2/share/cmake-3.24/Modules/FindHDF5.cmake:1063 (message):
   HDF5 was found, but a different variable was set which contains the
   location of the `hdf5::hdf5` library.
 Call Stack (most recent call first):
   cmake/HDF5Config.cmake:8 (find_package)
   CMakeLists.txt:116 (include)

 CMake Error at .../cmake/3.24.2/share/cmake-3.24/Modules/FindHDF5.cmake:1063 (message):
   HDF5 was found, but a different variable was set which contains the
   location of the `hdf5::hdf5_fortran` library.
 Call Stack (most recent call first):
   cmake/HDF5Config.cmake:8 (find_package)
   CMakeLists.txt:116 (include)

**Comment by pnavaro** [comment]
Hi Katharina, this one is tricky 😅 . I set the debug mode in the hdf5 cmake config. Could you try to see if we get more information...

**Comment by kkormann** [comment]
Thanks, @pnavaro. I run a fresh build after pulling main but the error message did not change...

**Comment by pnavaro** [comment]
Perharps there some details in CMakeCache.txt file. Check the variable FIND_PACKAGE_MESSAGE_DETAILS_HDF5

**Comment by kkormann** [comment]
Yes, what it finds is strange:

`FIND_PACKAGE_MESSAGE_DETAILS_HDF5:INTERNAL=[/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so][cfound components: C Fortran ][v1.10.8()]`

**Comment by pnavaro** [comment]
Seems an internal problem since it happens during the find_package call ...  I tried something with last commit

**Comment by pnavaro** [comment]
could you give me your cmake version also, there is a lot of changes between [3.10](https://cmake.org/cmake/help/v3.10/module/FindHDF5.html) and [3.27](https://cmake.org/cmake/help/v3.27/module/FindHDF5.html)

**Comment by pnavaro** [comment]
You can try 
cmake -DHDF5_USE_STATIC_LIBRARIES=ON

**Comment by pnavaro** [comment]
You can try to unset 
HDF5_USE_STATIC_LIBRARIES 
perhaps there are present on your system and there is only dynamic version

**Comment by pnavaro** [comment]
Another hint,  if you use env modules on your cluster try

module display hdf5

if there is any weird variable or path set.

**Comment by kkormann** [comment]
> You can try to unset HDF5_USE_STATIC_LIBRARIES perhaps there are not present on your system and there is only dynamic version

This is the solution, thanks a lot!
